### PR TITLE
`Development`: Add JavaDocs to UtilServices and Factories [16]

### PR DESCRIPTION
**Pull request by DominikRemo**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

Improve the readability of UtilServices and Factories by adding JavaDoc to the helper methods

### Description

Added documentation to following files:
- [x] post/ConversationUtilService.java
- [x] post/ConversationFactory.java

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/post/ConversationUtilService.java [hunk 1]
```diff
@@ -409,13 +592,27 @@ public void addReactionForUserToPost(String login, Post post) {
...
+     */
     public void addReactionForUserToAnswerPost(String login, AnswerPost answerPost) {
         Reaction reaction = ConversationFactory.createReactionForUser(userUtilService.getUserByLogin(login));
         reaction.setAnswerPost(answerPost);
         answerPostRepository.save(answerPost);
         reactionRepository.save(reaction);
     }
 
+    /**
+     * Creates and saves a Channel, a Post and a Reaction. Also creates and saves an AnswerPost with a Reaction.
```
Comment (JanaNF) [inline src/test/java/de/tum/in/www1/artemis/post/ConversationUtilService.java:609]: Again, "Also ..." --> "It also ..."

**Review by JanaNF — CHANGES_REQUESTED** [review]
Added some comments. You used quite a lot of "passive" sentences, but I didn't marked them because I am not sure if this is relevant here. If you consider changing this I would suggest using Grammarly.

File: src/test/java/de/tum/in/www1/artemis/post/ConversationUtilService.java [hunk 2]
```diff
@@ -258,13 +310,28 @@ private List<Post> createBasicPosts(Course courseContext, CourseWideContext[] co
...
+     */
     public Post createBasicPost(PlagiarismCase plagiarismCase, String userPrefix) {
         Post postToAdd = ConversationFactory.createBasicPost(0, userUtilService.getUserByLoginWithoutAuthorities(String.format("%s%s", userPrefix + "instructor", 1)));
         postToAdd.setPlagiarismCase(plagiarismCase);
         postToAdd.getPlagiarismCase().setExercise(null);
         return postRepository.save(postToAdd);
     }
 
+    /**
+     * Creates and saves 3 Posts for the given Conversation.
```
Comment (xHadie) [inline src/test/java/de/tum/in/www1/artemis/post/ConversationUtilService.java:328]: Maybe you could elaborate on this a bit more. What are those 3 posts and why are they generated?

**Review by xHadie — APPROVED** [review]
Overall, looks good to me. Jana already mentioned most formal things. The methods are already pretty self-explonatory and the variable descriptions are helpful. Added one minor suggestion.

File: src/test/java/de/tum/in/www1/artemis/post/ConversationUtilService.java (hunk 2 above)
Comment (DominikRemo) [reply src/test/java/de/tum/in/www1/artemis/post/ConversationUtilService.java]: I tried to be more descriptive and updated the documentation.

**Review by reschandreas — APPROVED** [review]
lgtm

**Pull request merged**